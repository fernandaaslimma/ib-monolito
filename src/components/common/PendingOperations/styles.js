import styled, { css } from "styled-components";
import { rem, remFontSize } from "../../../styles/tools";
import { white, darkGreen, gray200 } from "../../../styles/settings";

export const InProgressMovements = styled.div`
  background: linear-gradient(0deg, ${white}, ${white});
  border-radius: ${rem(4)};
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: ${rem(26)} ${rem(18)};
  margin: ${rem(19)} 0 ${rem(27)} 0;

  ${({ mode }) =>
    mode === "desktop" &&
    css`
      justify-content: center;
      cursor: pointer;
    `};
  ${({ withoutIcon }) =>
    withoutIcon === true &&
    css`
      padding: 0 ${rem(8)} ${rem(8)} ${rem(8)};
      margin: 0;
    `};
`;

export const ClicableItem = styled.div`
  cursor: pointer;
`;

export const HowManyMovements = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(14)};
  line-height: 125%;

  letter-spacing: ${rem(0.45)};

  color: ${darkGreen};

  ${({ mode }) =>
    mode === "desktop" &&
    css`
      padding: 0 ${rem(18)} 0 ${rem(18)};
    `};
  ${({ withoutIcon }) =>
    withoutIcon === true &&
    css`
      color: ${gray200};
    `};
`;

export const RotateComponent = styled.div`
  margin-right: ${rem(-7)};
  ${({ isRotate }) =>
    isRotate &&
    css`
      transform: rotate(180deg);
    `}

  ${({ angle }) =>
    angle &&
    css`
      transform: rotate(${angle}deg);
    `}
`;
