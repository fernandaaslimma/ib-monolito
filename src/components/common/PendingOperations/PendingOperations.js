import React, { Fragment, useState } from "react";
import Icon from "react-bocombbm-components/dist/Icon";

import { neutral200 } from "../../../styles/settings";
import AnimatedBottonSheet from "../../common/AnimatedBottomSheet";

import {
  InProgressMovements,
  HowManyMovements,
  RotateComponent,
  ClicableItem
} from "./styles";

function PendingOperations({
  mode,
  children,
  boxTitle,
  bottomSheetTitle,
  withoutIcon,
  printContext = false,
  dataTest
}) {
  const [bottomSheetState, changeBottomSheetState] = useState(false);

  return (
    <Fragment>
      <InProgressMovements
        data-test={`${dataTest}`}
        mode={mode}
        onClick={() => changeBottomSheetState(true)}
        withoutIcon={withoutIcon}
      >
        {!withoutIcon && (
          <Icon type="ClockFilled" width={20} height={20} color={neutral200} />
        )}
        <HowManyMovements mode={mode} withoutIcon={true}>
          {boxTitle}
        </HowManyMovements>
        <RotateComponent angle={270}>
          <ClicableItem>
            <Icon type="Arrow" height={20} width={20} color={neutral200} />
          </ClicableItem>
        </RotateComponent>
      </InProgressMovements>

      {!printContext && (
        <AnimatedBottonSheet
          isOpen={bottomSheetState}
          velocity={0.2}
          fullHeight
          dataTest={"animatedBottomSheet"}
          head={{ title: bottomSheetTitle, close: true, icon: "BorderedClose" }}
          onClickInBack={() => changeBottomSheetState(false)}
        >
          {children}
        </AnimatedBottonSheet>
      )}
    </Fragment>
  );
}

export default PendingOperations;
