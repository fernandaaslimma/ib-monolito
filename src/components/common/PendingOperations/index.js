import PendingOperations from "./PendingOperations";

export default PendingOperations;
