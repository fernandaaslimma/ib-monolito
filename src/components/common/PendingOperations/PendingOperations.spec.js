import React from "react";
import { render, screen, configure, fireEvent } from "@testing-library/react";
import PendingOperations from "./PendingOperations";

configure({ testIdAttribute: "data-test" });
jest.mock("../../../utils/i18n", () => ({
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false
}));

describe("PendingOperations component", () => {
  it("should match snapshot", () => {
    expect(render(<PendingOperations />)).toMatchSnapshot();
  });
  it("should clcik in progress movements", () => {
    const { rerender } = render(
      <PendingOperations dataTest="testing" printContext={false} />
    );
    const progressMovements = screen.getByTestId("testing");
    expect(fireEvent.click(progressMovements)).toBeTruthy();
    rerender(<PendingOperations dataTest="testing" printContext={false} />);
    const animatedBottomSheet = screen.getByTestId(
      "animatedBottomSheet-onClickInBack"
    );
    expect(fireEvent.click(animatedBottomSheet)).toBeTruthy();
  });
});
