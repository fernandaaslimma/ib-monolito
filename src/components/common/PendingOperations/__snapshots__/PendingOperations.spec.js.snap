// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingOperations component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: visible;"
  >
    <div>
      <div
        class="sc-jTzLTM iWfwOH"
        data-test="undefined"
      >
        <span
          class="sc-bdVaJa fmJszq"
          color="#3976cf"
          height="20"
          spacing="[object Object]"
          width="20"
        >
          <svg
            fill-rule="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 0C8.02219 0 6.08879 0.58649 4.4443 1.6853C2.79981 2.78412 1.51809 4.3459 0.761209 6.17317C0.00433284 8.00043 -0.193701 10.0111 0.192152 11.9509C0.578004 13.8907 1.53041 15.6725 2.92894 17.0711C4.32746 18.4696 6.10929 19.422 8.0491 19.8079C9.98891 20.1937 11.9996 19.9957 13.8268 19.2388C15.6541 18.4819 17.2159 17.2002 18.3147 15.5557C19.4135 13.9112 20 11.9778 20 10C20 8.68678 19.7413 7.38642 19.2388 6.17317C18.7363 4.95991 17.9997 3.85752 17.0711 2.92893C16.1425 2.00035 15.0401 1.26375 13.8268 0.761205C12.6136 0.258658 11.3132 0 10 0ZM14 11H10C9.73479 11 9.48043 10.8946 9.2929 10.7071C9.10536 10.5196 9 10.2652 9 10V6C9 5.73478 9.10536 5.48043 9.2929 5.29289C9.48043 5.10536 9.73479 5 10 5C10.2652 5 10.5196 5.10536 10.7071 5.29289C10.8946 5.48043 11 5.73478 11 6V9H14C14.2652 9 14.5196 9.10536 14.7071 9.29289C14.8946 9.48043 15 9.73478 15 10C15 10.2652 14.8946 10.5196 14.7071 10.7071C14.5196 10.8946 14.2652 11 14 11Z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="sc-jzJRlG jHpSRK"
        />
        <div
          class="sc-cSHVUG hLyTNB"
        >
          <div
            class="sc-fjdhpX gsvsfq"
          >
            <span
              class="sc-bdVaJa fmJszq"
              color="#3976cf"
              height="20"
              spacing="[object Object]"
              width="20"
            >
              <svg
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M 3.0325556 9.478114 l 3.947 -4.211 L 10.925555 9.478114 c 0.239 0.255 0.628 0.255 0.88 0.0 c 0.239 -0.255 0.239 -0.684 0.0 -0.939 L 7.419556 3.8721104 c -0.24 -0.255 -0.629 -0.255 -0.88 0.0 L 2.1645555 8.5521145 c -0.239 0.256 -0.239 0.685 0.0 0.94 c 0.239 0.254 0.629 0.254 0.868 -0.014 z"
                  fill="currentColor"
                  fill-rule="nonzero"
                  transform="matrix(-0.9996729 0.025575541 -0.025575541 -0.9996729 14.138712 13.182392)"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jTzLTM iWfwOH"
      data-test="undefined"
    >
      <span
        class="sc-bdVaJa fmJszq"
        color="#3976cf"
        height="20"
        spacing="[object Object]"
        width="20"
      >
        <svg
          fill-rule="none"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 0C8.02219 0 6.08879 0.58649 4.4443 1.6853C2.79981 2.78412 1.51809 4.3459 0.761209 6.17317C0.00433284 8.00043 -0.193701 10.0111 0.192152 11.9509C0.578004 13.8907 1.53041 15.6725 2.92894 17.0711C4.32746 18.4696 6.10929 19.422 8.0491 19.8079C9.98891 20.1937 11.9996 19.9957 13.8268 19.2388C15.6541 18.4819 17.2159 17.2002 18.3147 15.5557C19.4135 13.9112 20 11.9778 20 10C20 8.68678 19.7413 7.38642 19.2388 6.17317C18.7363 4.95991 17.9997 3.85752 17.0711 2.92893C16.1425 2.00035 15.0401 1.26375 13.8268 0.761205C12.6136 0.258658 11.3132 0 10 0ZM14 11H10C9.73479 11 9.48043 10.8946 9.2929 10.7071C9.10536 10.5196 9 10.2652 9 10V6C9 5.73478 9.10536 5.48043 9.2929 5.29289C9.48043 5.10536 9.73479 5 10 5C10.2652 5 10.5196 5.10536 10.7071 5.29289C10.8946 5.48043 11 5.73478 11 6V9H14C14.2652 9 14.5196 9.10536 14.7071 9.29289C14.8946 9.48043 15 9.73478 15 10C15 10.2652 14.8946 10.5196 14.7071 10.7071C14.5196 10.8946 14.2652 11 14 11Z"
            fill="currentColor"
          />
        </svg>
      </span>
      <div
        class="sc-jzJRlG jHpSRK"
      />
      <div
        class="sc-cSHVUG hLyTNB"
      >
        <div
          class="sc-fjdhpX gsvsfq"
        >
          <span
            class="sc-bdVaJa fmJszq"
            color="#3976cf"
            height="20"
            spacing="[object Object]"
            width="20"
          >
            <svg
              viewBox="0 0 14 14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 3.0325556 9.478114 l 3.947 -4.211 L 10.925555 9.478114 c 0.239 0.255 0.628 0.255 0.88 0.0 c 0.239 -0.255 0.239 -0.684 0.0 -0.939 L 7.419556 3.8721104 c -0.24 -0.255 -0.629 -0.255 -0.88 0.0 L 2.1645555 8.5521145 c -0.239 0.256 -0.239 0.685 0.0 0.94 c 0.239 0.254 0.629 0.254 0.868 -0.014 z"
                fill="currentColor"
                fill-rule="nonzero"
                transform="matrix(-0.9996729 0.025575541 -0.025575541 -0.9996729 14.138712 13.182392)"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
